src/main/java/com/jfrog/Utils.java

```diff
@@ -0,0 +1,121 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author yahavi
+ */
+public class Utils {
+    private static final ObjectMapper mapper = createMapper();
+
+    /**
+     * Create an Object Mapper used to deserialize the dependency tree to a JSON file.
+     *
+     * @return Object Mapper.
+     */
+    public static ObjectMapper createMapper() {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        return mapper;
+    }
+
+    /**
+     * Create the dependency tree for the given Maven project.
+     *
+     * @param dependencyGraphBuilder - A helper library that takes the project and session as inputs and constructs a Maven dependency tree
+     * @param session                - Maven execution session
+     * @param project                - The Maven "project" we are currently working on, specifically as a submodule
+     * @return Maven dependency tree
+     * @throws DependencyGraphBuilderException if some of the dependencies could not be resolved.
+     */
+    public static MavenDepTreeResults createDependencyTree(DependencyGraphBuilder dependencyGraphBuilder, MavenSession session,
+                                                           MavenProject project) throws DependencyGraphBuilderException {
+        ProjectBuildingRequest buildingRequest =
+                new DefaultProjectBuildingRequest(session.getProjectBuildingRequest());
+        buildingRequest.setProject(project);
+        DependencyNode root = dependencyGraphBuilder.buildDependencyGraph(buildingRequest, null);
+        Map<String, MavenDependencyNode> nodes = new HashMap<>();
+        populateDependencyMap(root, nodes);
+        return new MavenDepTreeResults(getGavString(root.getArtifact()), nodes);
+    }
+
+    /**
+     * Recursively populate the dependency nodes map from the given dependency tree.
+     *
+     * @param node  - The dependency tree of a Maven project
+     * @param nodes - Nodes map to populate
+     */
+    static void populateDependencyMap(DependencyNode node, Map<String, MavenDependencyNode> nodes) {
+        Artifact artifact = node.getArtifact();
+        MavenDependencyNode mavenDependencyNode = nodes.get(getGavString(artifact));
+        if (mavenDependencyNode == null) {
+            // Node does not exist in the Map - add it
+            mavenDependencyNode = new MavenDependencyNode(artifact.getScope());
+            nodes.put(getGavString(artifact), mavenDependencyNode);
+        } else {
+            // Node exists in the map - add the scope
+            mavenDependencyNode.addConfiguration(artifact.getScope());
```

Comment:
Shouldn't this line be outside of the `else` clause? Meaning, shouldn't we add the configuration anyway, even if the node has just been added to the map?

Reply:
2 lines above it puts the configuration in the constructor